src/app/teacher/apply-tags-button/apply-tags-button.component.ts

```diff
@@ -19,21 +21,37 @@ export class ApplyTagsButtonComponent implements OnInit {
   constructor(private dialog: MatDialog, private projectTagService: ProjectTagService) {}
 
   ngOnInit(): void {
-    this.retrieveUserTags();
     this.subscribeToTagUpdated();
     this.subscribeToNewTag();
     this.subscribeToTagDeleted();
   }
 
+  ngOnChanges(): void {
+    this.retrieveUserTags();
+  }
```

Comment:
Why do we need to retrieve the user tags each time selectedProjects changes? Shouldn't the tags stay the same?

Reply:
If you modify the tags in a different tab it can change, but that's not very likely to happen so I've made the change.